terraform/route53/dns/main.tf

```diff
@@ -18,16 +33,27 @@ resource "aws_route53_record" "dns-control" {
 resource "aws_route53_record" "dns-worker" {
   count = "${var.worker_count}"
   zone_id = "${var.hosted_zone_id}"
-  name = "${var.short_name}-worker-${format("%03d", count.index+1)}.${var.domain}"
-  records = ["${element(split(\",\", var.worker_ips), count.index)}"]
+  name = "${var.short_name}-worker-${format("%03d", count.index+1)}.node.${var.domain}"
+  records = ["${element(split(",", var.worker_ips), count.index)}"]
+  type = "A"
+  ttl = 60
+}
+
+# group records
+resource "aws_route53_record" "dns-control-group" {
+  count = "${var.control_count}"
+  zone_id = "${var.hosted_zone_id}"
+  name = "${var.control_subdomain}${var.subdomain}.${var.domain}"
+  records = ["${elements(split(",", var.control_ips), count.index)}"]
   type = "A"
   ttl = 60
 }
 
-resource "aws_route53_record" "dns-worker-haproxy" {
+resource "aws_route53_record" "dns-wildcard" {
+  count = "${var.edge_count}"
   zone_id = "${var.hosted_zone_id}"
-  name = "*.${var.short_name}-lb.${var.domain}"
-  records = ["${split(\",\", var.worker_ips)}"]
+  name = "*${var.subdomain}.${var.domain}"
+  records = ["${elements(split(",", var.edge_ips), count.index)}"]
```

Comment:
yeah... I'm not set up to test this provider, so I just tried not to make mistakes and hoped someone would catch them in review. Fortunately, you did. Thanks!